From 3a99c7fa58e418874f8158fa81a41cf18644d955 Mon Sep 17 00:00:00 2001
From: Patrik Stromdahl <6065kc@gmail.com>
Date: Sun, 28 May 2017 08:58:44 +0200
Subject: [PATCH 3/7] Terminal: Fix keyboard Ctrl- and ALT-key input.

Makes the Terminal app handle keyboard Ctrl-
and ALT-modifiers correctly.

Change-Id: Ie3d4feb072172496aff3a7e6d544d3e9683f15fb
---
 src/com/android/terminal/TerminalKeys.java | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/com/android/terminal/TerminalKeys.java b/src/com/android/terminal/TerminalKeys.java
index 7b12edf..9850f52 100644
--- a/src/com/android/terminal/TerminalKeys.java
+++ b/src/com/android/terminal/TerminalKeys.java
@@ -158,7 +158,14 @@ public class TerminalKeys {
     }
 
     public int getCharacter(KeyEvent event) {
-        int c = event.getUnicodeChar();
+        int c;
+
+        if (event.isShiftPressed()) {
+            c = event.getUnicodeChar(KeyEvent.META_SHIFT_LEFT_ON | KeyEvent.META_SHIFT_ON);
+        } else {
+            c = event.getUnicodeChar(0);
+        }
+
         // TODO: Actually support dead keys
         if ((c & KeyCharacterMap.COMBINING_ACCENT) != 0) {
             Log.w(TAG, "Received dead key, ignoring");
-- 
2.17.1

